ui/imports/shared/popups/CommunityIntroDialog.qml

```diff
@@ -18,64 +18,147 @@ import SortFilterProxyModel 0.2
     required property var permissionsModel  // id, key, permissionType, holdingsListModel, channelsListModel, isPrivate, tokenCriteriaMet
     required property var assetsModel
     required property var collectiblesModel
 
-    required property bool requirementsCheckPending
-
     property var keypairSigningModel
 
+    property var oldSharedAddresses: []
+    onOldSharedAddressesChanged: d.reEvaluateModels()
+    property string oldAirdropAddress: ""
+    onOldAirdropAddressChanged: d.reEvaluateModels()
+
     property var getCurrencyAmount: function (balance, symbol){}
 
+    property var canProfileProveOwnershipOfProvidedAddressesFn: function(addresses) { return false }
+
+    readonly property bool profileProvesOwnershipOfSelectedAddresses: {
+        d.selectedSharedAddressesMap // needed for binding
+        let obj = d.getSelectedAddresses()
+        return root.canProfileProveOwnershipOfProvidedAddressesFn(obj.addresses)
+    }
+
+    readonly property bool allAddressesToRevealBelongToSingleKeypairAndProfileKeypairCanNotProveOwnershipForThem: {
+        let keyUids = new Set()
+        for (let [key, value] of d.selectedSharedAddressesMap) {
+            keyUids.add(value.keyUid)
+        }
+        return keyUids.size === 1 && !keyUids.has(userProfile.keyUid)
+    }
+
     signal prepareForSigning(string airdropAddress, var sharedAddresses)
     signal joinCommunity()
-    signal signSharedAddressesForAllNonKeycardKeypairs()
+    signal editRevealedAddresses()
+    signal signProfileKeypairAndAllNonKeycardKeypairs()
     signal signSharedAddressesForKeypair(string keyUid)
     signal cancelMembershipRequest()
     signal sharedAddressesUpdated(var sharedAddresses)
 
     width: 640 // by design
     padding: 0
-    stackTitle: root.accessType === Constants.communityChatOnRequestAccess ? qsTr("Request to join %1").arg(name) : qsTr("Welcome to %1").arg(name)
+    stackTitle: root.accessType === Constants.communityChatOnRequestAccess ?
+                    qsTr("Request to join %1").arg(root.communityName)
+                  : qsTr("Welcome to %1").arg(root.communityName)
 
     rightButtons: [d.shareButton, finishButton]
 
     finishButton: StatusButton {
-        text: root.isInvitationPending ?
-                  qsTr("Cancel Membership Request")
-                : root.accessType === Constants.communityChatOnRequestAccess?
-                       qsTr("Prove ownership")
-                     : qsTr("Join %1").arg(root.name)
-
+        text: {
+            if (root.isInvitationPending) {
+                  return qsTr("Cancel Membership Request")
+            } else if (root.accessType === Constants.communityChatOnRequestAccess) {
+                if (d.selectedSharedAddressesCount === d.totalNumOfAddressesForSharing) {
+                    return qsTr("Share all addresses to join")
+                }
+                return qsTr("Share %n address(es) to join", "", d.selectedSharedAddressesCount)
```

Comment:
```suggestion
                return qsTr("Share %n address(s) to join", "", d.selectedSharedAddressesCount)
```

Yes, really has to be that way :)

Reply:
How does these plurals work? I can't understand it with your suggestion :thinking:

Reply:
Qt expects the ending to be exactly `(s)`; it then has to be translated, even for English to display the correct sentence